src/plugins/legacy/polygonRoi/polygonEventFilter.cpp

```diff
@@ -432,26 +651,145 @@ void polygonEventFilter::activateRepulsor(bool state)
+    {
+        int label = colorList.indexOf(manager->getColor()) + 1;
+
+        auto ctr = manager->getContoursAsNodes();
+        medTagContours contoursTag;
+
+        QString name = QString("label-%1").arg(label);
+        QString labelName = ( name == manager->getName() )?"undefined":manager->getName();
+        contoursTag.setLabelName(labelName);
+        contoursTag.setContourNodes(ctr);
+        contoursData.append(contoursTag);
+
+        auto pd = manager->getContoursAsPolyData(label);
+        append->SetInputDataByNumber(num, pd);
+        num++;
+    }
+    append->Update();
+    vtkMetaDataSet *outputDataSet = vtkMetaSurfaceMesh::New();
+    outputDataSet->SetDataSet(append->GetOutput());
+
+
+    saveContoursAsMedAbstractData(outputDataSet, contoursData);
+    return;
+
+}
+
+
+void polygonEventFilter::loadContours(medAbstractData *data)
+{
+    medContours *contours = dynamic_cast<medContours*>(data);
+    QVector<medTagContours> &tagContoursSet = contours->getTagContoursSet();
+
+    if ( managers.size()==colorList.size() )
+    {
+        qDebug()<<"unable to create new manager ";
+        return;
+    }
+
+    for (medTagContours &tagContours : tagContoursSet)
+    {
+        if ( managers.size()==colorList.size() )
+        {
+            qDebug()<<"unable to create new manager ";
+            return;
+        }
+        int label = findAvailableLabel();
+        if ( label == -1 )
+            return;
+        QString labelName = (tagContours.getLabelName() == "undefined")?QString("label-%1").arg(label):tagContours.getLabelName();
+        medTagRoiManager *manager = addManagerToList(label, labelName);
+        manager->loadContours(tagContours.getContourNodes());
+    }
+}
+
+int polygonEventFilter::findAvailableLabel()
 {
-    QList<UChar3ImageType::Pointer> masks;
-    for (medLabelManager *manager : managers)
+    QList<QColor> usedColors;
+    for ( medTagRoiManager* manager : managers )
     {
-        savedMask(manager);
+        usedColors.append(manager->getColor());
+    }
+    for (QColor color : colorList)
+    {
+        if (!usedColors.contains(color))
+        {
+            return colorList.indexOf(color);
+        }
     }
+    return -1;
 }
 
 void polygonEventFilter::manageTick()
 {
-    for (medLabelManager *manager : managers)
+    medIntParameterL *slicingParameter = nullptr;
+    for( auto interactor : qobject_cast<medAbstractLayeredView*>(currentView)->layerInteractors(0))
+    {
+        if ((interactor->identifier() == "medVtkViewItkDataImageInteractor") ||
+                (interactor->identifier() == "medVtkViewItkDataImage4DInteractor"))
+        {
+            for (auto parameter : interactor->linkableParameters())
+            {
+                if (parameter->name() == "Slicing")
+                {
+                    slicingParameter = qobject_cast<medIntParameterL*>(parameter);
+                    break;
+                }
+            }
+            break;
```

Comment:
is this break needed?

Reply:
I copied this code from another toolbox.
If slicing parameter is found in linkableParameters list, you must break in the layerInteractors for loop too.